File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if(nums.length < 1)
            return 0;
        
        
        Set set = new HashSet();
        int result = 1;
        
        for(int num : nums)
            set.add(num);
        
        for(int num : nums)
        {
            
            if(!set.contains(num + 1))
            {
                int answer = 1;
                while(set.contains(num - 1))
                {
                    answer++;
                    num--;
                }
                result = Math.max(answer, result);
            }
        }
        
        return result;
    }
}

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if(not nums):
            return 0
        
        
        existingSet = set()

        for num in nums:
            existingSet.add(num)

        longest = 0
        currentLongest = 1
        for num in existingSet:
            if num - 1 not in existingSet:
                current = num
                while current+1 in existingSet:
                    currentLongest += 1
                    current += 1
                longest = max(longest, currentLongest)
                currentLongest = 1
        return longest
            


        